api/api/V1/Processing.ts

```diff
@@ -352,10 +367,193 @@ export default function (app: Application, baseUrl: string) {
     },
   );
 
+  /**
+   * @api {post} /api/v1/:id/tracksAndTags Add tracks and tags to a recording
+   * @apiName PostTracksAndTags
+   * @apiGroup Processing
+   *
+   * Requires super-admin user credentials
+   *
+   * @apiParam {JSON} data Data which defines the tracks and tags (type specific).
+   * @apiParam {Number} AlgorithmId Database ID of the Tracking algorithm details retrieved from
+   * (#FileProcessing:Algorithm) request
+   *
+   * @apiUse V1ResponseSuccess
+   * @apiSuccess {int[]} trackIds of the newly created track.
+   *
+   * @apiUse V1ResponseError
+   *
+   */
+  app.post(
+    `${apiUrl}/:id/tracksAndTags`,
+    extractJwtAuthorisedSuperAdminUser,
+    validateFields([
+      idOf(param("id")),
+      body("data").custom(jsonSchemaOf(ApiMinimalTracksRequestSchema)),
+      idOf(body("algorithmId")),
+    ]),
+    parseJSONField(body("data")),
+    async (request: Request, response: Response, next: NextFunction) => {
+      const recording = await Recording.findByPk(request.params.id);
+      if (!recording) {
+        return next(
+          new AuthorizationError(
+            `Could not find a Recording with an id of '${request.params.id}'`,
+          ),
+        );
+      }
+      const data = response.locals.data;
+      const trackIds: number[] = [];
+      const tracks = [];
+      for (const trackData of data) {
+        const isMasked = await isTrackMasked(
+          recording,
+          trackData,
+          request.body.algorithmId,
```

Comment:
`isTrackMasked` doesn't seem to use the algorithmId at all, so this arg should be removed.